trin-bridge/src/bridge.rs

```diff
@@ -0,0 +1,580 @@
+    SszNone,
+};
+use trin_types::execution::receipts::Receipts;
+use trin_types::jsonrpc::params::Params;
+use trin_types::jsonrpc::request::JsonRequest;
+use trin_utils::bytes::hex_encode;
+use trin_validation::accumulator::MasterAccumulator;
+use trin_validation::constants::{EPOCH_SIZE as EPOCH_SIZE_USIZE, MERGE_BLOCK_NUMBER};
+use trin_validation::oracle::HeaderOracle;
+
+pub struct Bridge {
+    pub mode: BridgeMode,
+    pub portal_clients: Vec<HttpClient>,
+    pub header_oracle: HeaderOracle,
+    pub epoch_acc_path: PathBuf,
+}
+
+// todo: calculate / test optimal saturation delay
+const HEADER_SATURATION_DELAY: u64 = 10; // seconds
+const LATEST_BLOCK_POLL_RATE: u64 = 5; // seconds
+const BACKFILL_THREAD_COUNT: usize = 8;
+const EPOCH_SIZE: u64 = EPOCH_SIZE_USIZE as u64;
+
+impl Bridge {
+    // Devops nodes don't have websockets available, so we can't actually poll the latest block.
+    // Instead we loop on a short interval and fetch the latest blocks not yet served.
+    // We only need one thread for this mode
+    pub async fn launch_latest(&self) {
+        let mut block_index = get_latest_block_number().await.expect(
+            "Error launching bridge in latest mode. Unable to get latest block from provider.",
+        );
+        sleep(Duration::from_secs(LATEST_BLOCK_POLL_RATE)).await;
+        loop {
+            let latest_block = match get_latest_block_number().await {
+                Ok(val) => val,
+                Err(msg) => {
+                    warn!("error getting latest block, skipping iteration: {msg:?}");
+                    sleep(Duration::from_secs(LATEST_BLOCK_POLL_RATE)).await;
+                    continue;
+                }
+            };
+            if latest_block > block_index {
+                let block_range_to_gossip = Range {
+                    start: block_index,
+                    end: latest_block + 1,
+                };
+                info!("Discovered new blocks to gossip: {block_range_to_gossip:?}");
+                let full_headers = match Bridge::get_headers(&block_range_to_gossip).await {
+                    Ok(val) => val,
+                    Err(msg) => {
+                        warn!("error getting headers, skipping iteration: {msg:?}");
+                        block_index = latest_block + 1;
+                        sleep(Duration::from_secs(LATEST_BLOCK_POLL_RATE)).await;
+                        continue;
+                    }
+                };
+                // epoch index is 0 here bc it's irrelevant for post-merge headers
+                let mut gossip_batch = GossipBatch::new(block_range_to_gossip, 0, full_headers);
+                if let Err(msg) = self.gossip_headers(&mut gossip_batch).await {
+                    warn!("error gossiping headers, skipping iteration: {msg:?}");
+                    block_index = latest_block + 1;
+                    sleep(Duration::from_secs(LATEST_BLOCK_POLL_RATE)).await;
+                    continue;
```

Comment:
Rather than having to remember to sleep before every continue, maybe put the sleep at the top of the loop and drop them here?